sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+  <version>12.0.0-preview.2</version>
+</dependency>
+```
+
+## Generate SAS token
+SAS token generation has moved from helper classes to the core clients themselves.
+See section [SAS Token](#sastoken) to learn how to use them as a credential.
+Not all fields being set in these samples are necessary.
+At minimum, only an expiry time and permission set are necessary to create a SAS.
+
+In v10/v11, to generate the SAS token, you needed an instance of ServiceSASSignatureValues to fluently build out all it's options. This could them generate SASQueryParameters which contained the actual URL query string to attach to your URL. You also needed to provide container and blob names when necessary, as well as the `SharedKeyCredential` to sign the token with.
+```java
+// specify token properties
+ServiceSASSignatureValues value = new ServiceSASSignatureValues()
+BlobSASPermission p = new BlobSASPermission()
+        .withRead(true)
+        .withWrite(true)
+        .withCreate(true)
+        .withDelete(true)
+        .withAdd(true)
+value.withPermissions(p.toString())
+        .withStartTime(OffsetDateTime.now().minusDays(1))
+        .withExpiryTime(OffsetDateTime.now().plusDays(1))
+        .withContainerName(containerName)
+        .withBlobName(blobName)
+IPRange ipR = new IPRange()
+        .withIpMin("0.0.0.0")
+        .withIpMax("255.255.255.255")
+value.withIpRange(ipR)
+        .withProtocol(SASProtocol.HTTPS_ONLY)
+        .withCacheControl("cache")
+        .withContentDisposition("disposition")
+        .withContentEncoding("encoding")
+        .withContentLanguage("language")
+        .withContentType("type")
+
+// build the token
+SASQueryParameters sasToken = value.generateSASQueryParameters(sharedKeyCredentails)
+```
+
+In v12, SAS tokens are generated off the client to the resource you wish to generate the SAS for.
+The resource path handled automatically because of this, and the `SharedKeyCredential` authenticating the client is also used automatically to sign the SAS. See [building a client](#buildclient) for how to get an authenticated client.
+```java
+// specify token properties
+String identifier = "identifier";
+BlobSASPermission permissions = new BlobSASPermission()
+        .read(true)
+        .create(true)
+        .delete(true)
+        .write(true);
+OffsetDateTime startTime = OffsetDateTime.now().minusDays(1);
+OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+IPRange ipRange = new IPRange()
+        .ipMin("0.0.0.0")
+        .ipMax("255.255.255.255");
+SASProtocol sasProtocol = SASProtocol.HTTPS_HTTP;
+String cacheControl = "cache";
+String contentDisposition = "disposition";
+String contentEncoding = "encoding";
+String contentLanguage = "language";
+String contentType = "type";
+String version = Constants.HeaderConstants.TARGET_STORAGE_VERSION;
+
+// build the token
+String sasToken = blobClient.generateSAS(identifier, permissions, expiryTime, startTime, version, sasProtocol,
```

Comment:
This method was removed for GA. I am keen to add something like this back, but I seem to recall some pushback. Are we ok adding the method back with this signature? Or will this need to change?